server.js

```diff
@@ -7881,6 +7885,81 @@ cache(function(data, match, sendBadge, request) {
   });
 }));
 
+// Vaadin Directory Integration
+camp.route(/^\/vaadin-directory\/(star|status|rating|rc|lv|ld|maturity)\/(.*).(svg|png|gif|jpg|json)$/, cache(function (data, match, sendBadge, request) {
+  var type = match[1]; // Field required
+  var urlIdentifier = match[2]; // Name of repository
+  var format = match[3]; // Format
+  // API URL which contains also authentication info
+  var apiUrl = 'https://vaadin.com/vaadincom/directory-service/components/search/findByUrlIdentifier?projection=summary&urlIdentifier=' + urlIdentifier;
+
+  // Set left-side text to 'Vaadin-Directory' by default
+  var badgeData = getBadgeData("Vaadin Directory", data);
+  request(apiUrl, function(err, res, buffer) {
+    if (checkErrorResponse(badgeData, err, res)) {
+      sendBadge(format, badgeData);
+      return;
+    }
+
+    try {
+      var data = JSON.parse(buffer);
+      // Round the rating to 1 points decimal
+      var rating = ( Math.round(data.averageRating * 10) / 10 ).toFixed(1);
+      var ratingCount = data.ratingCount;
+      var lv = data.latestAvailableRelease.name.toLowerCase();
+      var ld = data.latestAvailableRelease.publicationDate;
+      switch (type) {
+        case 'star': // Stars
+          badgeData.text[0] = getLabel('rating', data);
+          badgeData.text[1] = starRating(rating);
+          badgeData.colorscheme = floorCountColor(rating, 2, 3, 4);
+          break;
+        case 'status': //published or not
+          var isPublished = data.status.toLowerCase();
+          if (isPublished === 'published') {
+            badgeData.text[1] = "published";
+            badgeData.colorB = '#00b4f0';
+          } else {
+            badgeData.text[1] = "unpublished";
+          }
+          break;
+        case 'rating': // ratings
+        badgeData.text[0] = getLabel('rating', data);
+          if (!isNaN(rating)) {
+            badgeData.text[1] = rating + '/5';
+            badgeData.colorscheme = floorCountColor(rating, 2, 3, 4);
+          }
+          break;
+        case 'rc': // rating counts
+          badgeData.text[0] = getLabel('rating count', data);
+          if (ratingCount && ratingCount != 0) {
+            badgeData.text[1] = metric(data.ratingCount) + ' total';
+            badgeData.colorscheme = floorCountColor(data.ratingCount, 5, 50, 500);
+          }
+          break;
+        case 'lv': // latest version
+          badgeData.text[0] = getLabel('latest ver', data);
+          badgeData.text[1] = lv;
+          badgeData.colorscheme = "blue";
+          break;
+        case 'ld':
+          badgeData.text[0] = getLabel('latest release date', data);
+          // TODO: Shorten the date
```

Comment:
A relative date I guess. The options for `getLabel` are not too many though, it only contains `last day, same day, last week and else as MMMM YYYY`. It would be really nice to have how many weeks ago if the weeks are less than a month.

Reply:
Okay no worries, If you could remove the `// TODO` i think this should be good for merge!

Also if you in the future do decide you want the message to be `3 weeks ago`, `1 month ago` etc feel free to create a new function in `/lib/text-formatters` :smile:

Reply:
Haha okay, got it. I guess I'll do that sometime later. Quick question, so do I edit the current function `formatDate` or just make a new one?

Anyhow, I just removed the comment, if you could merge it now so we got it working in the beginning of next week. That would be a blast for us.

Reply:
Probably best to make a new function as to not change the badges that currently use it, unless it also makes sense for those badges.